### PR TITLE
4454 - Save chat text to file

### PR DESCRIPTION
**Pull request by Cattlesquat**

### CODE REVIEW

File: vassal-app/src/main/java/VASSAL/build/module/Chatter.java [hunk 1]
```diff
@@ -197,10 +206,74 @@ public void keyReleased(KeyEvent e) {
...
+          name = name + ".html";
+          fc.setSelectedFile(new File(name));
+        }
+
+        final File outputFile = fc.getSelectedFile();
+        if (outputFile == null) {
+          return;
+        }
+
+        try (OutputStream fos = Files.newOutputStream(Path.of(outputFile.getPath()))) {
```
Comment (uckelman) [inline vassal-app/src/main/java/VASSAL/build/module/Chatter.java:233]: `outputFile.toPath()` will get you a `Path`.

File: vassal-app/src/main/java/VASSAL/build/module/Chatter.java [hunk 2]
```diff
@@ -197,10 +206,74 @@ public void keyReleased(KeyEvent e) {
...
+          fc.setSelectedFile(new File(name));
+        }
+
+        final File outputFile = fc.getSelectedFile();
+        if (outputFile == null) {
+          return;
+        }
+
+        try {
+          FileUtils.writeStringToFile(outputFile, doc.getText(0, doc.getLength()), Charset.forName("UTF-8")); //NON-NLS
```
Comment (uckelman) [inline vassal-app/src/main/java/VASSAL/build/module/Chatter.java:265]: Use `StandardCharsets.UTF_8`.

Comment (Cattlesquat) [reply vassal-app/src/main/java/VASSAL/build/module/Chatter.java]: fixed

File: vassal-app/src/main/java/VASSAL/build/module/Chatter.java (hunk 1 above)
Comment (Cattlesquat) [reply vassal-app/src/main/java/VASSAL/build/module/Chatter.java]: fixed

**Pull request merged**